src/redirect/server.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-undef */
+const express = require('express');
+const base64url = require('base64url');
+const crypto = require('crypto');
+const { PORT, PUBLIC_URL } = require('./config');
+const logger = require('./logger');
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+// create express app
+const app = express();
+app.use(express.json());
+app.listen(PORT, () => {
+  logger.info(`App listening at http://localhost:${PORT}`);
+});
+
+/**
+ * Creates random Id for QR code data that is POSTed to the server.
+ * @returns base 64 url encoded Id
+ */
+const randomId = () => base64url.encode(crypto.randomBytes(32));
+
+const QRs = new Map();
+
+/*
+POST QR data to server.
+
+If there is more than one file to share, server creates a Sharing Manifest JSON file with
+array of high-entropy shared file URLS.
+
+Generates JSON payload (returned as the response) containing a URL for the authorization server.
+The QR will be constructed from this JSON QR payload.
+*/
+app.post('/qr', async (req, res) => {
+  logger.info('Sending POST request to /qr');
+  console.log(process.env.PORT);
+  const qrId = randomId();
+  // set mapping of ID to object containing QR details
+  QRs.set(qrId, {
+    id: qrId,
+    request: req.body,
+    claims: []
+  });
+
+  // generate and return url for the authorization server
+  res.json({
+    url: `${PUBLIC_URL}/qr/${qrId}/claim`
+  });
+});
+
+/*
+POST to .url property of the JSON payload to redeem shclink.
+
+If the request is valid, server returns 301 status and Location header redirecting
+client to URL where file/Sharing Manifest is exposed.
+*/
```

Comment:
On Postman, I'm getting a `200` status, not `301`. And a `locations` property in the body, but no `Location` header.

Reply:
Makes sense, thanks for clarifying.

Reply:
Looking at the last test case you added clears things up as well!